thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/cube/additive/AdditiveDBClient.java

```diff
@@ -0,0 +1,97 @@
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.pinot.thirdeye.cube.additive;
+
+import com.google.common.base.Preconditions;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import org.apache.pinot.thirdeye.cube.data.dbrow.DimensionValues;
+import org.apache.pinot.thirdeye.cube.data.dbrow.Dimensions;
+import org.apache.pinot.thirdeye.cube.data.dbclient.CubeTag;
+import org.apache.pinot.thirdeye.cube.data.dbclient.BaseCubePinotClient;
+import org.apache.pinot.thirdeye.cube.data.dbclient.CubeSpec;
+import org.apache.pinot.thirdeye.datasource.cache.QueryCache;
+
+/**
+ * This class generates query requests to the backend database and retrieve the data for summary algorithm.
+ *
+ * The generated requests are organized the following tree structure:
+ *   Root level by GroupBy dimensions.
+ *   Mid  level by "baseline" or "current"; The "baseline" request is ordered before the "current" request.
+ *   Leaf level by metric functions; This level is handled by the request itself, i.e., a request can gather multiple
+ *   metric functions at the same time.
+ * The generated requests are store in a List. Because of the tree structure, the requests belong to the same
+ * timeline (baseline or current) are located together. Then, the requests belong to the same GroupBy dimension are
+ * located together.
+ */
+public class AdditiveDBClient extends BaseCubePinotClient<AdditiveRow> {
+  private String metric;
+
+  /**
+   * Constructs a DB client to an additive metric.
+   *
+   * @param queryCache the query cache to Pinot DB.
+   */
+  public AdditiveDBClient(QueryCache queryCache) {
+    super(queryCache);
+  }
+
+  /**
+   * Sets the additive metric name.
+   *
+   * @param metric the additive metric name.
+   */
+  public void setMetric(String metric) {
+    this.metric = Preconditions.checkNotNull(metric);
+  }
+
+  @Override
+  protected List<CubeSpec> getCubeSpecs() {
+    List<CubeSpec> cubeSpecs = new ArrayList<>();
+
+    cubeSpecs.add(new CubeSpec(CubeTag.Baseline, metric, baselineStartInclusive, baselineEndExclusive));
+    cubeSpecs.add(new CubeSpec(CubeTag.Current, metric, currentStartInclusive, currentEndExclusive));
+
+    return cubeSpecs;
+  }
+
+  @Override
+  protected void fillValueToRowTable(Map<List<String>, AdditiveRow> rowTable, Dimensions dimensions,
+      List<String> dimensionValues, double value, CubeTag tag) {
+
+    if (Double.compare(0d, value) < 0 && !Double.isInfinite(value)) {
+      AdditiveRow row = rowTable.get(dimensionValues);
+      if (row == null) {
+        row = new AdditiveRow(dimensions, new DimensionValues(dimensionValues));
+        rowTable.put(dimensionValues, row);
+      }
+      switch (tag) {
+        case Baseline:
+          row.setBaselineValue(value);
+          break;
+        case Current:
+          row.setCurrentValue(value);
+          break;
+        default:
```

Comment:
should it throw an exception in this case?

Reply:
The default is no-op because tags are shared across additive and ratio metrics. If I throw an exception, then I won't be able to support both cases.

Reply:
So the `AdditiveDBClient` will also work with ratio metrics?

Reply:
No, it won't. There will be another `DBClient` called `RatioDBClient`. I see your point now; will amend it.